### PR TITLE
Use pmem library to create pmem devices and run xfstests

### PR DESCRIPTION
**Pull request by harish-24**
Patch uses pmem library to create pmem devices and run xfstests suite

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
**nvdimm_log.yaml**
```
JOB LOG    : /home/harish/avocado-fvt-wrapper/results/job-2020-03-02T12.17-419f69a/job.log
 (1/2) fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-log_scratch-log_test-fs_ext4-464c:  FAIL: One or more tests failed. Please check the logs. (4992.05 s)
 (2/2) fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-log_scratch-log_test-fs_xfs-7c87:  FAIL: One or more tests failed. Please check the logs. (9251.26 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 14244.23 s
```
**nvdimm.yaml**
```
JOB LOG    : /home/harish/avocado-fvt-wrapper/results/job-2020-03-02T22.04-faaca54/job.log
 (1/2) fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-fs_ext4-b619:  FAIL: One or more tests failed. Please check the logs. (2541.44 s)
 (2/2) fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-fs_xfs-7bb0:  FAIL: One or more tests failed. Please check the logs. (4141.45 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 6683.90 s
```

File: fs/xfstests.py [hunk 1]
```diff
@@ -106,6 +101,16 @@ def setUp(self):
         self.disk_mnt = self.params.get('disk_mnt', default='/mnt/loop_device')
         self.dev_type = self.params.get('type', default='loop')
         self.fs_to_test = self.params.get('fs', default='ext4')
+        if self.dev_type == 'nvdimm':
+            packages.extend(['ndctl', 'parted'])
+            if self.detected_distro.name == 'rhel':
+                packages.extend(['daxctl'])
+
+        for package in packages:
+            if not sm.check_installed(package) and not sm.install(package):
```
Comment (PraveenPenguin) [inline fs/xfstests.py]: club this to above `https://github.com/avocado-framework-tests/avocado-misc-tests/blob/af2fbb8363bcd6d4b9709c69adf77f77ab7ddcd1/fs/xfstests.py#L59` as based on nvdim is dev type extend package list there it self

File: fs/xfstests.py [hunk 2]
```diff
@@ -129,6 +134,59 @@ def setUp(self):
                 else:
                     self.cancel('Need %s GB to create loop devices' % check)
             self._create_loop_device(base_disk, loop_size, mount)
+        elif self.dev_type == 'nvdimm':
+            self.test_dev = self.params.get('disk_test', default=None)
```
Comment (PraveenPenguin) [inline fs/xfstests.py]: get this assignment as global and change value inside loop that seems me a good way

File: fs/xfstests.py [hunk 3]
```diff
@@ -211,11 +270,11 @@ def setUp(self):
                     self._create_test_list(self.share_exclude, "shared",
                                            dangerous=False)
         if self.detected_distro.name is not 'SuSE':
-            process.run('useradd 123456-fsgqa', sudo=True)
-            process.run('useradd fsgqa', sudo=True)
+            process.run('useradd 123456-fsgqa', sudo=True, ignore_status=True)
```
Comment (PraveenPenguin) [inline fs/xfstests.py]: why `ignore_status=True` if user creation fails test should not run right ?

File: fs/xfstests.py.data/nvdimm_log.yaml [hunk 4]
```diff
@@ -12,12 +13,13 @@ setup:
             #share_exclude: '1-300'
         fs_xfs:
             fs: 'xfs'
-            mkfs_opt: '-b size=65536 -s size=4096'
+            mkfs_opt: '-b size=65536 -s size=512 -m reflink=0'
```
Comment (PraveenPenguin) [inline fs/xfstests.py.data/nvdimm_log.yaml]: why this ymal changes i feel this is just a template user can have there owned YAML file

File: fs/xfstests.py (hunk 1 above)
Comment (harish-24) [reply fs/xfstests.py]: Sure

File: fs/xfstests.py (hunk 2 above)
Comment (harish-24) [reply fs/xfstests.py]: As we have different way of handling each disk_type, i think it should not matter.

File: fs/xfstests.py (hunk 3 above)
Comment (harish-24) [reply fs/xfstests.py]: Okay, let me add a check to warn if the command fails.

File: fs/xfstests.py.data/nvdimm_log.yaml (hunk 4 above)
Comment (harish-24) [reply fs/xfstests.py.data/nvdimm_log.yaml]: These options are necessary for nvdimm type disk, which makes sure mount always works.

**Comment by harish-24** [comment]
@PraveenPenguin Pushed with changes, Thanks.

**Pull request merged**